### PR TITLE
list directory if index not found

### LINKED ISSUE #275: 'miniserve --index index.html' gives 404 when index.html is not present in a directory.
## Steps to reproduce

```sh
mkdir -p public/sub
echo 'Hello There' > public/sub/index.html
cd public # public does not contain an index.html
miniserve -i 8080 --index index.html .
```

Open the browser, go to http://localhost:8080/ and http://localhost:8080/sub/.

## Expected behavior

* http://localhost:8080/ should display a directory list.
* http://localhost:8080/sub/ should display content of `public/sub/index.html`.

## Actual behavior

* http://localhosy:8080/ is not found (404).

### PR DESCRIPTION
**Pull request by aliemjay**
closes #275 
supersedes #484

### CODE REVIEW

File: tests/serve_request.rs [hunk 1]
```diff
@@ -220,3 +220,14 @@ fn serves_requests_custom_index_notice(tmpdir: TempDir, port: u16) -> Result<(),
 
     Ok(())
 }
+
+#[rstest]
+#[case(server(&["--index", FILES[0]]))]
+#[case(server(&["--index", "does-not-exist.html"]))]
+fn index_fallback_to_listing(#[case] server: TestServer) -> Result<(), Error> {
+    // If index file is not found, show directory listin instead.
```
Comment (svenstaro) [inline tests/serve_request.rs:228]: Typo: `listing`.

File: src/main.rs [hunk 2]
```diff
@@ -299,29 +304,22 @@ fn configure_app(app: &mut web::ServiceConfig, conf: &MiniserveConfig) {
...
-            if show_hidden {
-                files = actix_files::Files::new(&full_route, path)
-                    .show_files_listing()
-                    .use_hidden_files();
-            } else {
-                files = actix_files::Files::new(&full_route, path).show_files_listing();
-            }
-
+            let files = actix_files::Files::new(&full_route, path);
+            let files = match &conf.index {
```
Comment (svenstaro) [inline src/main.rs:313]: Would be great if you could document this whole block of logic while you're here. :)

**Review by svenstaro — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 2 above)
Comment (aliemjay) [reply src/main.rs:315]: I have more upcoming refactoring for this, which should make it clearer.

Comment (svenstaro) [reply src/main.rs:315]: Alright, that works for me, too, just make sure to keep track of it. :P

**Review by svenstaro — APPROVED** [review]

**Comment by svenstaro** [comment]
Short and sweet!

**Pull request merged**